service/manager.go

```diff
@@ -267,24 +283,8 @@ func (s *manager) addModule(provider ModuleProvider, options ...ModuleOption) er
 			zap.Any("roles", moduleWrapper.scopedHost.Roles()),
 		)
 	}
-	return s.addModuleWrapper(moduleWrapper)
-}
-
-// StartAsync service is used as a non-blocking the call on service start. StartAsync will
-// return the call to the caller with a Control to listen on channels
-// and trigger manual shutdowns.
-func (s *manager) StartAsync() Control {
-	return s.start()
-}
-
-// Start service is used for blocking the call on service start. Start will block the
-// call and yield the control to the service lifecyce manager. Start will not yield back
-// the control to the caller, so no code will be executed after calling Start()
-func (s *manager) Start() {
-	s.start()
-
-	// block until forced exit
-	s.WaitForShutdown(nil)
+	s.moduleWrappers = append(s.moduleWrappers, moduleWrapper)
+	return nil
 }
 
 func (s *manager) start() Control {
```

Comment:
nit: could you remove the error on the following line?
I ran [grinder](https://github.com/rsc/grind) on our code and it was the only issue.